pkg/operator/ceph/object/user/controller.go

```diff
@@ -225,25 +343,52 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 		StringData: secrets,
 		Type:       k8sutil.RookType,
 	}
-	k8sutil.SetOwnerRef(&secret.ObjectMeta, &c.ownerRef)
 
-	_, err = context.Clientset.CoreV1().Secrets(u.Namespace).Create(secret)
+	return secret
+}
+
+func (r *ReconcileObjectStoreUser) reconcileCephUserSecret(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
+	// Generate Kubernetes Secret
+	secret := r.generateCephUserSecret(cephObjectStoreUser)
+
+	// Set owner ref to the object store user object
+	err := controllerutil.SetControllerReference(cephObjectStoreUser, secret, r.scheme)
 	if err != nil {
-		return errors.Wrapf(err, "failed to save user %s secret", u.Name)
+		return reconcile.Result{}, errors.Wrapf(err, "failed to set owner reference for ceph object user %q secret", secret.Name)
 	}
-	logger.Infof("created user %s", u.Name)
-	return nil
+
+	// Create Kubernetes Secret
+	err = opcontroller.CreateOrUpdateObject(r.client, secret)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create or update ceph object user %q secret", secret.Name)
+	}
+
+	// _, err = r.context.Clientset.CoreV1().Secrets(cephObjectStoreUser.Namespace).Create(secret)
```

Comment:
nit: can remove this dead code now?

Reply:
yes yes sorry I have a local commit that fixes this :)